pkg/operator/controller/controller_router_deployment.go

```diff
@@ -73,20 +73,37 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, routerImage strin
 	deployment.Name = name.Name
 	deployment.Namespace = name.Namespace
 
-	if deployment.Labels == nil {
-		deployment.Labels = map[string]string{}
+	deployment.Labels = map[string]string{
+		// associate the deployment with the ingresscontroller
+		manifests.OwningClusterIngressLabel: ci.Name,
```

Comment:
Should this be using `IngressControllerDeploymentLabel`?

Reply:
Not sure, this is the same as before,  and deploymentLabel is for ingress.operator.openshift.io/ingress-controller-deployment — we might need to audit our labels generallly

Reply:
Meant for this comment to apply to controller_internal_service.go

Reply:
And no, this was to associate the deployment with the ingresscontroller, which is the intent of the label... IngressControllerDeploymentLabel was meant to be a way to associate other stuff with the deployment itself. The naming is probably poor

Reply:
What is the intended use of `IngressControllerDeploymentLabel`? I was assuming it was, "I have an ingresscontroller and need the label I would use to identify it."

Reply:
Sorry, just saw your last comment (GitHub didn't show it when I submitted mine last comment).  I see now, `IngressControllerDeploymentLabel` identifies the deployment, and we need a label that identifies the ingresscontroller (for which we do not yet have a function).  Please disregard my preceding comments.